apps/epic-web/src/pages/index.tsx

```diff
@@ -69,6 +69,7 @@ const Index: NextPage<{
   const {redeemableCoupon, RedeemDialogForCoupon, validCoupon} = useCoupon(
     commerceProps?.couponFromCode,
     {
+      id: product.productId,
```

Comment:
The fix for epic-web is to pass in the `productId` wherever this is called which gets passed down through several components and an API call and eventually makes its way into a `productIds` array for the `validateCoupon` call.